lmfit/lineshapes.py

```diff
@@ -302,6 +302,26 @@ def expsine(x, amplitude=1.0, frequency=1.0, shift=0.0, decay=0.0):
     return amplitude*sin(x*frequency + shift) * exp(-x*decay)
 
 
+def thermal_distribution(x, amplitude=1.0, center=0.0, kT=1.0, kind=-1):
+    """Return a thermal distribution function
+    kind = -1 (default) is the Bose-Einstein distribution
+    kind = 0 is the Maxwell-Boltzmann distribution
+    kind = 1 is the Fermi-Dirac distribution
+
+    thermal_distribution(x, amplitude=1.0, center=0.0, kBT=1.0, kind=-1):
+       = 1/(amplitude*exp((x - center)/kT) + kind)
+
+    Notes:
+    - kT should be defined in the same units as x
+    -The Boltzmann constant is kB = 8.617e-5 eV/K
+    set kBT<0 to implement the energy loss convention common in scattering
+    research.
+
+    see http://hyperphysics.phy-astr.gsu.edu/hbase/quantum/disfcn.html
+    """
+    return real(1/(amplitude*exp((x - center)/kT) + kind + tiny*1j))
```

Comment:
I don't understand what the purpose is of first generating a complex array by adding ```tiny*1j``` and the immediately take the real part of that... can you elaborate on why this is needed/useful?

Reply:
In the bose form the function crosses over from +inf to -inf at center=0. The trick makes the function go through zero.

Reply:
bose diverges from + to - inf through x=center. This makes the value zero since it arguably goes through zero.